numpy/tests/test_scripts.py

```diff
@@ -92,7 +93,14 @@ def try_f2py_commands(cmds):
         version = sys.version_info
         major = str(version.major)
         minor = str(version.minor)
+        f2py_success = try_f2py_commands(('f2py',))
         f2py_cmds = ('f2py', 'f2py' + major, 'f2py' + major + '.' + minor)
-        success = try_f2py_commands(f2py_cmds)
-        msg = "Warning: not all of %s, %s, and %s are found in path" % f2py_cmds
-        assert_(success == 3, msg)
+        all_success = try_f2py_commands(f2py_cmds)
+        if all_success == 3:
+            return
+        elif f2py_success == 1:
+            msg = "f2py fails"
```

Comment:
There is something special about the windows linux subsystem as this test passes on appveyor. I don't know how to check for the linux subsystem, so it would be better to simply check for `f2py`, or maybe for the presence of at least one of the options, which is what the test used to do.

Reply:
@charris it fails on multiple other systems, see gh-10016

I think this test does need to be simplified. The logic here is that if all 3 commands are not found, it checks just for `f2py` and if that is found then the test does not fail. I think that's the main thing to agree on first. The rest can be tweaked as needed.

Reply:
Yep, I think just checking for `f2py` is the way to go. That used to fail for other systems also, but maybe it is better now that we are using entry points. @tylerjereddy Does that simplified test work for you?

Reply:
On my mac with this diff on master:
```diff
--- a/numpy/tests/test_scripts.py
+++ b/numpy/tests/test_scripts.py
@@ -92,7 +92,7 @@ def test_f2py():
         version = sys.version_info
         major = str(version.major)
         minor = str(version.minor)
-        f2py_cmds = ('f2py', 'f2py' + major, 'f2py' + major + '.' + minor)
+        f2py_cmds = ('f2py',)
         success = try_f2py_commands(f2py_cmds)
-        msg = "Warning: not all of %s, %s, and %s are found in path" % f2py_cmds
-        assert_(success == 3, msg)
+        msg = "Warning: %s not found in path" % f2py_cmds
+        assert_(success == 1, msg)
```

From `python runtests.py -t "numpy/tests/test_scripts.py" -- -s` I get a failure and `f2py` itself gives:
```
Traceback (most recent call last):
  File "/Users/treddy/miniconda3/envs/numpy_dev_py36/bin/f2py", line 11, in <module>
    load_entry_point('numpy==1.16.0.dev0+025bbdd', 'console_scripts', 'f2py')()
  File "/Users/treddy/miniconda3/envs/mda_dev_py3/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg/pkg_resources/__init__.py", line 565, in load_entry_point
  File "/Users/treddy/miniconda3/envs/mda_dev_py3/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg/pkg_resources/__init__.py", line 2598, in load_entry_point
  File "/Users/treddy/miniconda3/envs/mda_dev_py3/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg/pkg_resources/__init__.py", line 2258, in load
  File "/Users/treddy/miniconda3/envs/mda_dev_py3/lib/python3.6/site-packages/setuptools-27.2.0-py3.6.egg/pkg_resources/__init__.py", line 2264, in resolve
  File "/Users/treddy/.local/lib/python3.6/site-packages/numpy-1.16.0.dev0+5f2a5ae-py3.6-macosx-10.9-x86_64.egg/numpy/f2py/__main__.py", line 4, in <module>
    from f2py2e import main
ModuleNotFoundError: No module named 'f2py2e'
```

The modified test and `f2py` itself can be made to work if I adjust my path on mac os to check the build path first, before site-packages, with: 
`export PATH=/Users/treddy/github_projects/numpy_linked_worktree/build/testenv/bin:$PATH`
I did something similar to get shippable builds all green.

Once the path search order was fixed, I went back and reversed the change to the unit test noted above and the original test passed on my mac, so...

I'm pretty sure @pv previously commented somewhere about the difficulty of getting the `f2py` path stuff working just right in a portable fashion.